bitbots_time_constraint/src/bitbots_time_constraint/check_constraints.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# -*- coding:utf-8 -*-
+
+import rospy
+import rostopic
+import rospkg
+import yaml
+import os
+
+class AutoSubscriber:
+    """
+    A class which automatically subscribes to a topic as soon as it becomes available and saves the time when messages come in.
+    """
+
+    def __init__(self, topic):
+        """
+        :param topic: Topic to subscribe to
+        :type topic: str
+        """
+        self.topic = topic
+        self.latest_timestamp = None
+
+        self.__subscriber = None        # type: rospy.Subscriber
+        self.__subscribe()
+
+    def __subscribe(self, backoff=1.0):
+        """
+        Try to subscribe to the set topic
+        :param backoff: How long to wait until another try
+        """
+        data_class, _, _ = rostopic.get_topic_class(self.topic)
+        if data_class is not None:
+            # topic is known
+            self.__subscriber = rospy.Subscriber(self.topic, data_class, self.__message_callback, queue_size=1, tcp_nodelay=True)
+            rospy.loginfo('Subscribed to topic {}'.format(self.topic))
+
+        else:
+            # topic is not yet known
+            rospy.loginfo('Topic {} is not yet known. Retrying in {} seconds'.format(self.topic, int(backoff)))
+            rospy.Timer(
+                rospy.Duration(int(backoff)),
+                lambda event: self.__subscribe(backoff * 1.2),
+                oneshot=True
+            )
+    
+    def get_timestamp(self):
+        return self.latest_timestamp
+
+    def __message_callback(self, data):
+        self.latest_timestamp = rospy.Time.now()
+
+
+class ConstraintChecker:
+    def __init__(self):
+        rospack = rospkg.RosPack()
+        self.package_path = rospack.get_path('bitbots_time_constraint')
+
+        rospy.init_node("bitbots_time_constraint", anonymous=False)
+
+        config_path = os.path.join(self.package_path, "config/constraints.yaml")
+
+        with open(config_path, 'r') as stream:
+            try:
+                constraint_config = yaml.safe_load(stream)
+            except yaml.YAMLError as exc:
+                rospy.roserr("Config not found!!!!")
+                rospy.signal_shutdown("")
```

Comment:
Why? The ROS parameter server is able to read YAML files and you are using a launch file. Please use the ROS parameter server.